invenio_app_ils/circulation/api.py

```diff
@@ -77,6 +79,8 @@ def request_loan(
         raise PatronHasRequestOnDocumentError(patron_pid, document_pid)
     _validate_delivery(kwargs.get("delivery"))
 
+    transaction_user_pid = transaction_user_pid or current_user.id
```

Comment:
Instead of redefining `transaction_user_pid` we can put the condition where it is actually used.

Reply:
I guess this is more developer style: I personally prefer, in general, to have all inputs validation/sanitization on the top. If a variable is used multiple times (not sure this is the case), I would set this once and not multiple times, each time it is used.